### PR TITLE
Add starknet doc

### PR DESCRIPTION
**Pull request by LeJamon**
add documentation for the starknet sag.

### CODE REVIEW

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 1]
```diff
@@ -1,6 +1,135 @@
 ---
 id: SAG-starknet
 title: "SAG for Starknet"
---- 
+---
+
+# SAG for Starknet
+
+The **SAG for Starknet** implementation enables verification of Spontaneous Anonymous Group (SAG) signatures on the Starknet platform. This implementation leverages Starknet-specific optimizations to perform ring signature verification efficiently on-chain, supporting both SECP256K1 and ED25519 elliptic curves.
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:8]:
```suggestion
The **SAG for Starknet** implementation enables verification of Spontaneous Anonymous Group (SAG) signatures on the Starknet blockchain. This implementation leverages Starknet-specific optimizations to perform ring signature verification efficiently on-chain, supporting both SECP256K1 and ED25519 elliptic curves.
```

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 2]
```diff
@@ -1,6 +1,135 @@
...
+```typescript
+import {
+  RingSignature,
+  Curve,
+  CurveName,
+  Point,
+} from "@cypher-laboratory/alicesring-sag-starknet";
+
+const curve = new Curve(CurveName.SECP256K1);
+const ring: Point[] = []; // Your ring of public keys
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:33]: might be good to add 2 or 3 public keys so the user can see how they should be encoded (compressed to 33Bytes I guess)

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 3]
```diff
@@ -1,6 +1,135 @@
...
+  const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);
+
+  // Verify locally
+  console.log("Is signature valid?", signature.verify()); // Should output: true
+
+  const signature = RingSignature.sign(ring, signerPrivKey, message, curve);
+  const rawCallData = await signature.getCallData();
+  console.log(rawCallData);
+}
+ringSignForStarknet();
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:48]:
```suggestion

// Sign
const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);

// Verify locally
console.log("Is signature valid?", signature.verify()); // Should output: true

(async () => {
  // Get the signature data to send to the contract
  const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);
  const rawCallData = await signature.getCallData();
  console.log(rawCallData);
})();
```

- snippet was not valid (signature was defined twice in the same block and `signerPrivKey` was undefined)
- replaced the function by a simple async block (increase readability)
- only let the async call in the async block
- explained what is done in the `rawCallData` generation 

Can you also explain a little more the link between `rawCallData` and the contract (how is it used, if it is a function argument, explain whch one, ...). Maybe add a snippet where you generate a signature and verify it using the contract

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 4]
```diff
@@ -1,6 +1,135 @@
...
+  console.log("Is signature valid?", signature.verify()); // Should output: true
+
+  const signature = RingSignature.sign(ring, signerPrivKey, message, curve);
+  const rawCallData = await signature.getCallData();
+  console.log(rawCallData);
+}
+ringSignForStarknet(); 
+```
+
+**Important:** ringSignature.getCallData is an **async** function.
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:51]:
```suggestion
**Important:** ringSignature.getCallData() is an **async** function.
```

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 5]
```diff
@@ -1,6 +1,135 @@
...
+async function ringSignForStarknet() {
+  // Sign
+  const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);
+
+  // Verify locally
+  console.log("Is signature valid?", signature.verify()); // Should output: true
+
+  const signature = RingSignature.sign(ring, signerPrivKey, message, curve);
+  const rawCallData = await signature.getCallData();
+  console.log(rawCallData);
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:46]: can you add some context in the log ? So when someone run the code, they can easily know what this is.

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 6]
```diff
@@ -1,6 +1,135 @@
...
+    }
+}
+```
+
+### Using Voyager block explorer :
+
+You can also pass the raw call data to the deployed contract at the following address:
+
+**Starknet Contract Address**: [`0x0317Ce745DE0A65308A19f6e184Be43A17F20D1eAe04B7433d984Df1D571DAA4`](https://sepolia.voyager.online/contract/0x0317Ce745DE0A65308A19f6e184Be43A17F20D1eAe04B7433d984Df1D571DAA4#readContract) on Sepolia Starknet.
+
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:99]:
```suggestion
```

this is defined twice (see `Deployed address` section)

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 7]
```diff
@@ -1,6 +1,135 @@
...
+### Number of steps
+
+Verifying a SAG signature requires approximately 46,800 steps per member in the ring.
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:102]:
```suggestion
Verifying a SAG signature requires approximately 46,800 steps per public key in the ring.
```

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 8]
```diff
@@ -1,6 +1,135 @@
...
+| ---------------- | ------------------------------------------------------------------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Starknet-Sepolia | <img src="https://cryptologos.cc/logos/starknet-token-strk-logo.png?v=035" alt="Starknet" width="30"/> | [`0x0317Ce745DE0A65308A19f6e184Be43A17F20D1eAe04B7433d984Df1D571DAA4`](https://sepolia.voyager.online/contract/0x0317Ce745DE0A65308A19f6e184Be43A17F20D1eAe04B7433d984Df1D571DAA4) |
+
+## Contribution
+
+We welcome contributions! To contribute, follow these steps:
+
+1. Fork the [Repository for the signer](https://github.com/Cypher-Laboratory/Alice-s-Ring)
+   or the [Repository for the verifier](https://github.com/Cypher-Laboratory/Alice-s-Ring-Cairo-verifier) : Create a fork of the repository in your GitHub account.
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:115]: Why aren't them both in the `Alice's Ring` monorepo ?

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 9]
```diff
@@ -1,6 +1,135 @@
...
+  console.log(rawCallData);
+}
+ringSignForStarknet(); 
+```
+
+**Important:** ringSignature.getCallData is an **async** function.
+
+## Verifying a Ring Signature On-Chain
+
+To verify the ring signature on-chain, you can use the [SAG verifier in Cairo](https://github.com/Cypher-Laboratory/Alice-s-Ring-Cairo-verifier).
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:55]: why isn't this verifier in the monorepo ?

**Review by Elli610 — CHANGES_REQUESTED** [review]

File: packages/documentation/docs/SAG/SAG-for-starknet.md (hunk 9 above)
Comment (LeJamon) [reply packages/documentation/docs/SAG/SAG-for-starknet.md:56]: I don't know how scarb will manage the import of the package in a monorepo (as you pass the git repo to import). 
But we could try !

File: packages/documentation/docs/SAG/SAG-for-starknet.md (hunk 2 above)
Comment (LeJamon) [reply packages/documentation/docs/SAG/SAG-for-starknet.md]: done

File: packages/documentation/docs/SAG/SAG-for-starknet.md (hunk 5 above)
Comment (LeJamon) [reply packages/documentation/docs/SAG/SAG-for-starknet.md]: done

File: packages/documentation/docs/SAG/SAG-for-starknet.md (hunk 6 above)
Comment (LeJamon) [reply packages/documentation/docs/SAG/SAG-for-starknet.md:113]: this one refers to the action of making a call data to the block explorer (similar to this : https://garaga.gitbook.io/garaga/using-garaga-libraries-in-your-cairo-project/ec-multi-scalar-multiplication)

Comment (Elli610) [reply packages/documentation/docs/SAG/SAG-for-starknet.md:113]: can you explain this in the doc ?

File: packages/documentation/docs/SAG/SAG-for-starknet.md [hunk 10]
```diff
@@ -1,6 +1,148 @@
...
+
+const curve = new Curve(CurveName.SECP256K1);
+const message = "Hello, Alice's Ring!";
+
+// Define an array of compressed public keys
+const compressedPublickeys = [
+  "0316d7da70ba247a6a40bb310187e8789b80c45fa6dc0061abb8ced49cbe7f887f",
+  "0221869ca3ae33be3a7327e9a0272203afa72c52a5460ceb9f4a50930531bd926a",
+  "02337d6f577e66a21a7831c087c6836a1bae37086bf431400811ac7c6e96c8ccbb",
+];
```
Comment (Elli610) [inline packages/documentation/docs/SAG/SAG-for-starknet.md:40]:
```suggestion
const compressedPublickeys = [
  "0316d7da70ba247a6a40bb310187e8789b80c45fa6dc0061abb8ced49cbe7f887f",
  "0221869ca3ae33be3a7327e9a0272203afa72c52a5460ceb9f4a50930531bd926a",
  // ... 
];
```

**Review by Elli610 — APPROVED** [review]

File: packages/documentation/docs/SAG/SAG-for-starknet.md (hunk 6 above)
Comment (LeJamon) [reply packages/documentation/docs/SAG/SAG-for-starknet.md:113]: yep i will do a small video like in the garaga book

**Pull request merged**